### PR TITLE
feat: Integrate e2b.

### LINKED ISSUE #1245: [Feature Request] integrate `e2b`
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

https://github.com/e2b-dev/

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by willshang76**
## Description

Integrate the e2b interpreter into camel. https://github.com/e2b-dev/

## Motivation and Context

 `close #1245`

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Wendong-Fan** [comment]
Thanks @willshang76 , for the dependency issue, PR raised here: https://github.com/AgentOps-AI/agentops/pull/556

File: camel/interpreters/e2b_interpreter.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                    "further code execution."
+                )
+
+        if self._CODE_TYPE_MAPPING[code_type] is None:
+            execution = self._sandbox.run_code(code)
+        else:
+            execution = self._sandbox.run_code(
+                code=code, language=self._CODE_TYPE_MAPPING[code_type]
+            )
+        return execution.text
```
Comment (Wendong-Fan) [inline camel/interpreters/e2b_interpreter.py:111]: if execution failed, `execution.text` would be `None`, we should return the error message to user
`return execution.text if execution.text else str(execution.error)`

**Review by Wendong-Fan — COMMENTED** [review]
Thanks @willshang76 ! Left some comments below

File: camel/interpreters/e2b_interpreter.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        self,
+        code: str,
+        code_type: str,
+    ) -> str:
+        r"""Executes the given code in the e2b sandbox.
+
+        Args:
+            code (str): The code string to execute.
+            code_type (str): The type of code to execute (e.g., 'python',
+                'bash').
```
[the remark this answers is not in the document]
Comment (willshang76) [reply camel/interpreters/e2b_interpreter.py:79]: Actually the base class defined this abstract method as type str for this parameter. Therefore I think that we could leave it as this for this PR, we could create a separate issue to refactor this part. How do you think?

**Comment by lightaime** [comment]
What is `e2d` in the poetry?

**Comment by willshang76** [comment]
> What is `e2d` in the poetry?

e2b-code-interpreter

**Comment by lightaime** [comment]
There is a `e2d` as well. I think it is a typo

**Comment by willshang76** [comment]
> There is a `e2d` as well. I think it is a typo

The e2b-code-interpreter package actually depends on the e2b package. When I did poetry add e2b-code-interpreter; it also gets the e2b.

**Comment by lightaime** [comment]
It is `e2d` not `e2b`

File: camel/interpreters/e2b_interpreter.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                    "further code execution."
+                )
+
+        if self._CODE_TYPE_MAPPING[code_type] is None:
+            execution = self._sandbox.run_code(code)
+        else:
+            execution = self._sandbox.run_code(
+                code=code, language=self._CODE_TYPE_MAPPING[code_type]
+            )
+        return execution.text if execution.text else str(execution.error)
```
Comment (Wendong-Fan) [inline camel/interpreters/e2b_interpreter.py:118]: I found sometimes execution.text could return str `"None"`, which could make the check working not as expected, sometimes the code execution result is in `stdout`, could you help double check this part? Thanks @willshang76 

example result from execution:
```
Execution(Results: [], Logs: Logs(stdout: ['dwd\n'], stderr: []), Error: None)
```

**Pull request merged**